docs/taskruns.md

```diff
@@ -399,6 +401,56 @@ inside the `Pod`. Only the above command is affected. The `Pod's` description co
 denotes a "Failed" status and the container statuses correctly denote their exit codes
 and reasons.
 
+### Overriding Task Steps and Sidecars
+
+**([alpha only](https://github.com/tektoncd/pipeline/blob/main/docs/install.md#alpha-features))**
+**Warning: This feature is still under development and is not yet functional. Do not use it.**
+
+A TaskRun can specify `StepOverrides` or `SidecarOverrides` to override Step or Sidecar
+configuration specified in a Task.
+
+For example, given the following Task definition:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: image-build-task
+spec:
+  steps:
+    - name: build
+      image: gcr.io/kaniko-project/executor:latest
+  sidecars:
+    - name: logging
+      image: my-logging-image
+```
+
+An example TaskRun definition could look like:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: image-build-taskrun
+spec:
+  taskRef:
+    name: image-build-task
+  stepOverrides:
+    - name: build
+      resources:
+        requests:
+          memory: 1Gi
+  sidecarOverrides:
+    - name: logging
+      resources:
+        requests:
+          cpu: 100m
+        limits:
+          cpu: 500m
+```
+`StepOverrides` and `SidecarOverrides` must include the `name` field and may include `resources`.
+No other fields can be overridden.
```

Comment:
NIT: maybe worth mentioning that it is not possible to override resources for steps that don't have a name